### PR TITLE
[fix](compaction)Fix cluster key column duplicated in value columns group

### LINKED ISSUE #44609: [Bug] compaction failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

mater

### What's Wrong?

```
W 2024-11-25 01:29:37,712  9521 tablet.cpp:1850] failed to do cumulative compaction, tablet=1732457852810 : [INTERNAL_ERROR]InvertedIndexFileWriter::open attempted to insert a duplicate dir

        0#  doris::segment_v2::InvertedIndexFileWriter::_insert_directory_into_map(long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<doris::segment_v2::DorisFSDirectory>) at /var
/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/basic_string.h:658
        1#  doris::segment_v2::InvertedIndexFileWriter::open(doris::TabletIndex const*) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/shared_ptr_base.h:1149
        2#  doris::segment_v2::InvertedIndexColumnWriterImpl<(doris::FieldType)16>::open_index_directory() at /home/zcp/repo_center/doris_master/doris/be/src/util/expected.hpp:1986
        3#  doris::segment_v2::InvertedIndexColumnWriterImpl<(doris::FieldType)16>::init_bkd_index() at /home/zcp/repo_center/doris_master/doris/be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:175
        4#  doris::segment_v2::InvertedIndexColumnWriterImpl<(doris::FieldType)16>::init() at /home/zcp/repo_center/doris_master/doris/be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:139
        5#  doris::segment_v2::InvertedIndexColumnWriter::create(doris::Field const*, std::unique_ptr<doris::segment_v2::InvertedIndexColumnWriter, std::default_delete<doris::segment_v2::InvertedIndexColumnWriter> >*, doris::segment_v2::I
nvertedIndexFileWriter*, doris::TabletIndex const*) at /home/zcp/repo_center/doris_master/doris/be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:790
        6#  doris::segment_v2::ScalarColumnWriter::init() at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:502
        7#  doris::segment_v2::SegmentWriter::_create_column_writer(unsigned int, doris::TabletColumn const&, std::shared_ptr<doris::TabletSchema> const&) at /home/zcp/repo_center/doris_master/doris/be/src/common/status.h:502
        8#  doris::segment_v2::SegmentWriter::_create_writers(std::shared_ptr<doris::TabletSchema> const&, std::vector<unsigned int, std::allocator<unsigned int> > const&) at /home/zcp/repo_center/doris_master/doris/be/src/olap/rowset/seg
ment_v2/segment_writer.cpp:0
        9#  doris::segment_v2::SegmentWriter::init(std::vector<unsigned int, std::allocator<unsigned int> > const&, bool) at /home/zcp/repo_center/doris_master/doris/be/src/olap/rowset/segment_v2/segment_writer.cpp:304
        10# doris::VerticalBetaRowsetWriter<doris::BetaRowsetWriter>::add_columns(doris::vectorized::Block const*, std::vector<unsigned int, std::allocator<unsigned int> > const&, bool, unsigned int, bool) at /home/zcp/repo_center/doris_m
aster/doris/be/src/olap/rowset/vertical_beta_rowset_writer.cpp:0
        11# doris::Merger::vertical_compact_one_group(std::shared_ptr<doris::BaseTablet>, doris::ReaderType, doris::TabletSchema const&, bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, doris::vectorized::RowSourcesB
uffer*, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > > const&, doris::RowsetWriter*, long, doris::Merger::Statistics*, std::vector<unsigned int, std::allocator<unsigned int> >, lo
ng, doris::CompactionSampleInfo*) at /home/zcp/repo_center/doris_master/doris/be/src/olap/merger.cpp:294
        12# doris::Merger::vertical_merge_rowsets(std::shared_ptr<doris::BaseTablet>, doris::ReaderType, doris::TabletSchema const&, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > >
 const&, doris::RowsetWriter*, long, long, doris::Merger::Statistics*) at /home/zcp/repo_center/doris_master/doris/be/src/olap/merger.cpp:457
        13# doris::Compaction::merge_input_rowsets() at /home/zcp/repo_center/doris_master/doris/be/src/olap/compaction.cpp:194
        14# doris::CompactionMixin::execute_compact_impl(long) at /home/zcp/repo_center/doris_master/doris/be/src/olap/compaction.cpp:467
        15# doris::CompactionMixin::execute_compact() at /home/zcp/repo_center/doris_master/doris/be/src/olap/compaction.cpp:422
        16# doris::CumulativeCompaction::execute_compact() at /home/zcp/repo_center/doris_master/doris/be/src/olap/cumulative_compaction.cpp:0
        17# doris::Tablet::execute_compaction(doris::CompactionMixin&) at /home/zcp/repo_center/doris_master/doris/be/src/util/defer_op.h:37
        18# std::_Function_handler<void (), doris::StorageEngine::_submit_compaction_task(std::shared_ptr<doris::Tablet>, doris::CompactionType, bool)::$_0>::_M_invoke(std::_Any_data const&) at /home/zcp/repo_center/doris_master/doris/be/
src/olap/olap_server.cpp:1080
        19# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_master/doris/be/src/util/threadpool.cpp:0
        20# doris::Thread::supervise_thread(void*) at /var/local/ldb-toolchain/bin/../usr/include/pthread.h:563
        21# ?
        22# ?
```

### What You Expected?

Compact file normally.

### How to Reproduce?

1. create table with cluster key
```
CREATE TABLE test_index_compaction_unique_keys_arr (
                `id` int(11) NULL,
                `name` varchar(255) NULL,
                `hobbies` text NULL,
                `score` int(11) NULL,
                index index_name (name) using inverted,
                index index_hobbies (hobbies) using inverted properties("parser"="english"),
                index index_score (score) using inverted
            ) ENGINE=OLAP
            UNIQUE KEY(`id`)
            CLUSTER BY (`score`)
            COMMENT 'OLAP'
            DISTRIBUTED BY HASH(`id`) BUCKETS 1
            PROPERTIES ( 
                "replication_num" = "1",
                "disable_auto_compaction" = "true",
                "enable_unique_key_merge_on_write" = "true"
            );
```
2. insert data
```
INSERT INTO test_index_compaction_unique_keys_arr VALUES (1, "andy", "andy love apple", 100); 
INSERT INTO test_index_compaction_unique_keys_arr VALUES (1, "bason", "bason hate pear", 99); 
```
3. trigger full compaction
```
curl --max-time 10 -X POST http://10.16.10.6:8147/api/compaction/run?tablet_id=23859&compact_type=full
```
4. get error

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qidaye**
### What problem does this PR solve?

Issue Number: close #44609

Related pr: #40372

Problem Summary:

`key_columns` will be used after move.
This will cause skip in construction of `value_columns` useless.
cluster key  column will be duplicated added in value column group.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Review by mymeiyi — APPROVED** [review]

File: regression-test/suites/inverted_index_p0/array_contains/test_index_compaction_unique_keys_arr.groovy [hunk 1]
```diff
@@ -152,6 +152,7 @@ suite("test_index_compaction_unique_keys_arr", "array_contains_inverted_index")
                 index index_score (score) using inverted
             ) ENGINE=OLAP
             UNIQUE KEY(`id`)
+            CLUSTER BY (`score`)
```
Comment (airborne12) [inline regression-test/suites/inverted_index_p0/array_contains/test_index_compaction_unique_keys_arr.groovy:155]: give another case, not just from current.

**Comment by qidaye** [comment]
run buildall

**Review by airborne12 — DISMISSED** [review]
LGTM

**Comment by qidaye** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**